atom/src/validatebehavior.cpp

```diff
@@ -14,6 +16,34 @@
 using namespace PythonHelpers;
 
 
+bool validate_type_tuple_types( PyObject* type_tuple_types)
+{
+    if( PyTuple_Check( type_tuple_types ) )
+    {
+        int i, len;
+        len = PySequence_Size( type_tuple_types );
+        for (i = 0; i < len; i++) {
+            if( !PyType_Check( PyTuple_GET_ITEM( type_tuple_types, i) ) )
+            {
+                PyErr_Format(
+                    PyExc_TypeError,
+                    "Expected type or tuple of types. Got a tuple containing an instance of `%s` instead.",
+                    PyTuple_GET_ITEM( type_tuple_types, i)->ob_type->tp_name
+                );
+                return false;
+            }
+        }
+        return true;
+    }
+    if( !PyType_Check( type_tuple_types ) )
+    {
+        py_expected_type_fail( type_tuple_types, "type or tuple of types" );
+        return false;
+    }
+    return true;
```

Comment:
I will try to look into it.

Reply:
I'd say just leave it out, I can't say i've ever seen it used.

Reply:
The only case where it may cause an issue is for a class trying to not look like one, the way I understand the docs. So I agree it is a real corner case. I will proof read once more before merging.